### ISSUE TITLE
False positive with `clippy::if_then_some_else_none`

### ISSUE THREAD

**Issue by EFanZh** [rust-lang/rust-clippy#7870]
Lint name:

`clippy::if_then_some_else_none`

I tried this code:

```rust
#![warn(clippy::if_then_some_else_none)]

fn f(b: bool, v: Option<()>) -> Option<()> {
    if b {
        v?;

        Some(())
    } else {
        None
    }
}
```

I expected to see this happen: Clippy does not warn `clippy::if_then_some_else_none`.

Instead, this happened: Clippy  warns that code can be simplified.

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.58.0-nightly (91b931926 2021-10-23)
binary: rustc
commit-hash: 91b931926fd49fc97d1e39f2b8206abf1d77ce7d
commit-date: 2021-10-23
host: x86_64-pc-windows-msvc
release: 1.58.0-nightly
LLVM version: 13.0.0
```
